code/go/0chain.net/sharder/blockstore/cache.go

```diff
@@ -0,0 +1,253 @@
+package blockstore
+
+// Cache is a simple implementation using simple lru for storing block's minimal data.
```

Comment:
rocksdb has its own implementation of mem cache, so let's not implement new cache on top of rocksdb again. Basically if we write data to rocksdb and flush it, we want to make sure the data is persisted and won't lose data when we restart the node. For the cache implemented here, if we have default 100 cache buffer size and the sharder restarted unexpectedly, will all of those 100 blocks of data be persisted? if not, I suggest to avoid using the cache and use the rocksdb directly.

Reply:
@peterlimg The rocksdb cache is for its keys. Not sure if it could work in this case.
The uncompressed blocks are stored in a directory as specified in `yaml` file. Sharder can provide cache-size in config file and ready to go.

Blocks could be in Hard drive or in minio. Recently used block are stored uncompressed in cache directory and when the read request for block is made, it first checks cache directory.